op-batcher/batcher/channel_manager.go

```diff
@@ -71,6 +74,14 @@ func (s *channelManager) TxFailed(id txID) {
 	} else {
 		s.log.Warn("unknown transaction marked as failed", "id", id)
 	}
+
+	// If this channel has no submitted transactions, put the pending blocks back into the
+	// local saved blocks and reset this state so it can try to build a new channel.
+	if len(s.confirmedTransactions) == 0 && len(s.pendingTransactions) == 0 {
+		s.log.Info("Channel has no submitted transactions", "chID", s.pendingChannel.ID())
+		s.blocks = append(s.pendingChannel.Blocks(), s.blocks...)
+		s.clearPendingChannel()
+	}
```

Comment:
We could even explicitly speed up shutdown in this case like
```suggestion
	if s.closed && len(s.confirmedTransactions) == 0 && len(s.pendingTransactions) == 0 {
		s.log.Info("Channel has no submitted transactions, clearing for shutdown", "chID", s.pendingChannel.ID())
		s.Clear()
	}
```
wdyt @trianglesphere ?

Reply:
This actually identifies an interesting ordering issue:
If we mark the channel manager's only transaction as failed and then close it, we'll still have this pending frame. As a result, we'll still attempt to submit this one more time. The previous implementation would not be subject to this ordering problem, but would be more prone to unnecessary channel clearing.

Reply:
I realized that we can also handle this other case inside of `ChannelManager.Close()` in order to avoid unnecessary channel erasure as well as extra retries.